Feature/dropdown

simian114 opened the pull request:
- [x] 드롭다운 구현
- [x] useDetectOutsideClick 구현
  - 컴포넌트 외부 클릭을 감지
- [x] svg 다운

select 를 사용하면 디자인에 제약이 생길거같아서 직접구현했습니다.
사용예를 52e2378 커밋에 작성했습니다.

 props 로 넣어줄 값은 총 3개입니다.
1. selectedItem
2. onItemClick
3. options

options 는 [{print: 'print', data: 'data'}, {}, ... ] 의 형태를 갖습니다.
print값은 외부에 보여주는 값이고 data 는 선택 되었을 때 저장되는 값입니다.

seletedItem 은 dropdown 을 사용하는 부모컴포넌트에서 가지고 있는 state 입니다.
이 값은 options 의 data 값을 갖습니다.

onItemClick 은 드롭다운의 아이템을 선택했을 때 수행하는 콜백입니다.
seletedItem 의 setter 를 넣으면 됩니다.


Closes #17 #21 #24

simian114 committed:
[#18] feat: svg 아이콘 생성

simian114 committed:
[#19] feat: alabaster 색 추가

simian114 committed:
[#21] feat: useDetectOutsideClick 훅 구현

simian114 committed:
[#17] feat: 드롭다운 구현

드롭다운을 구현했습니다. props 로 넣어줄 값은 총 3개입니다.
1. selectedItem
2. onItemClick
3. options

options 는 [{print: 'print', data: 'data'}, {}, ... ] 의 형태를
갖습니다. print값은 외부에 보여주는 값이고 data 는 선택되었을 때
저장되는 값입니다.

seletedItem 은 dropdown 을 사용하는 부모컴포넌트에서 가지고 있는
state 입니다. 이 값은 options 의 data 값을 갖습니다.

onItemClick 은 드롭다운의 아이템을 선택했을 때 수행하는
콜백입니다. seletedItem 의 setter 를 넣으면 됩니다.

simian114 committed:
[#17] chore: dropdown 사용 예 작성

daeun-react commented on src/components/temp/DropdownTemp.tsx:
```diff
@@ -0,0 +1,28 @@
+const DropdownTemp: React.FC<any> = () => {
+  const [selectedOption, setSelectedOption] = useState<string>('');
+
+  useEffect(() => {
+    if (!selectedOption) {
+      console.log('카테고리를 선택해주세요!');
+    }
+    console.log(selectedOption);
+  }, [selectedOption]);
```
드랍다운 기능 구현 감사합니다!! 💯 🥇💯 
해당 부분은 머지 이후에 정렬에 대한 로직이 들어가는 부분일까요?! 
지금은 콘솔로그만 있어서 필요한 부분인지 확인차 문의드립니다! 😊

simian114 replied:
`DropdownTemp` 은 사용 예로 작성한 파일입니다!! 머지 하기 전에 지울게요 👍 👍

daeun-react replied:
아 예시 Temp 파일 이었군요! ㅎㅎ 사용하게 될때 참고해서 쓰도록 하겠습니다 😄😄

daeun-react commented on src/components/temp/DropdownTemp.tsx:
```diff
@@ -0,0 +1,28 @@
+    console.log(selectedOption);
+  }, [selectedOption]);
+
+  const ops: any = [{ print: '카테고리', data: '' }];
```
ops type에 any 대신 지정할 타입이 있을까요?🙄
저도 타입 넣는게 익숙하지 않아서 😂😂찾아보는 중이라 정답은 모르겠지만 😂😂
any 타입으로 되어 있는 부분 전체적으로 수정되면 좋을 것 같다는게 제 생각입니다!

simian114 replied:
이거도 사용예로 작성한거라 나중에 사용하실 때는 적절한 타입 지정하시면 될거같습니다! 👍

daeun-react replied:
네네, 타입은 사용 시에 고민해보겠습니다! 답변 감사합니다!👍👍

daeun-react reviewed (commented):
드랍다운 코드 잘 봤습니다! 🚀🚀🚀 
코드 리뷰 코멘트 남겨드립니다~ 확인 부탁드립니다!!

simian114 committed:
[#17] chore: 프리티어 적용

juneandkang reviewed (commented):
useDetectOutsideClick 구현해놓으신거 잘봤습니다! 👍👍👍
드롭다운뿐만 아니라 모달에서도 사용할수 있겠네요!

simian114 committed:
[#24] getRandomEnum 유틸함수 구현

simian114 committed:
[#24] feat: mockData 구현

해당 목 데이터는 localStorage 에 `todos` 키가 없을 때만 동작합니다.
만약 빈배열로 존재한다면, `mockData` 는 세팅되지 않습니다.

아무것도 없을 때 데이터 상으로 `mockData` 는 가져와지지만,
수정, 삭제, 추가 등 `todos` 를 변경하는 작업이 발생하지 않으면
`localStorage` 에 `set` 되지 않습니다.

juneandkang reviewed (approved):
목데이터 잘 짜주셨네요! 목데이터를 상수처럼 사용하는게 아니라 catergory나 priority를 랜덤으로 받아오는 방식은 처음 보는데 괜찮은 듯 합니다! 수고하셨어요 👍

irisdew commented:
상혁님 만들어주신 드롭다운 저도 활용해보겠습니다. 
활용 예시 임시로 만들어주신거 잘 커스텀해서 사용할게요 👍👍👍👍👀
mock data와 svg 파일 추가도 감사합니다
